app/src/main/java/com/spendster/presentation/addExpenses/AddExpensesPresenter.java

```diff
@@ -0,0 +1,66 @@
+package com.spendster.presentation.addExpenses;
+
+import android.content.Context;
+import android.content.res.Resources;
+
+import com.spendster.R;
+import com.spendster.data.entity.Expense;
+import com.spendster.presentation.authentication.SharedPreferencesUserStorage;
+
+import java.text.DecimalFormat;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.observers.DisposableSingleObserver;
+import io.reactivex.schedulers.Schedulers;
+
+public class AddExpensesPresenter {
+    private final Context context;
+    private final AddExpensesView addExpensesView;
+    private final ServerAddExpensesModel serverAddExpensesModel;
+    private final CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    public AddExpensesPresenter(AddExpensesView addExpensesView, ServerAddExpensesModel
+            serverAddExpensesModel, Context contex) {
+        this.addExpensesView = addExpensesView;
+        this.serverAddExpensesModel = serverAddExpensesModel;
+        this.context = contex;
+    }
+
+    public void save(double amount, String title, String note, String categoryId, long date) {
+        // validation
+        SharedPreferencesUserStorage sharedPreferencesUserStorage = new SharedPreferencesUserStorage(context);
```

Comment:
@SimonMaha It's better to pass base interface of SharedPreferencesUserStorage into constructor, instead of context